Agoda.Frameworks.Http/RandomUrlHttpClient.cs

```diff
@@ -111,6 +111,15 @@ public void Dispose()
         public Task<HttpResponseMessage> DeleteAsync(string url) =>
             SendAsync(url, uri => HttpClient.DeleteAsync(uri));
 
+        public async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            return await SendAsync(request.RequestUri.PathAndQuery, (url) =>
+            {
+                request.RequestUri = new Uri(url);
```

Comment:
Good trick for replacing URL, but it does not look straightforward. Please add unit test for that.

Additionally, please put this method into `#if !NET462` block. HttpRequestMessage cannot be reused for netframework.